base/linalg/symmetric.jl

```diff
@@ -117,8 +117,16 @@ A_mul_B!{T<:BlasComplex,S<:StridedMatrix}(C::StridedMatrix{T}, A::StridedMatrix{
 *(A::HermOrSym, B::HermOrSym) = full(A)*full(B)
 *(A::StridedMatrix, B::HermOrSym) = A*full(B)
 
-factorize(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+bkfact(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issym(A))
+factorize(A::HermOrSym) = bkfact(A)
```

Comment:
Okay, what I meant is simply to move the check to the Symmetric/Hermitian method. Something like 
```diff
diff --git a/base/linalg/dense.jl b/base/linalg/dense.jl
index ab2dac2..c39940f 100644
--- a/base/linalg/dense.jl
+++ b/base/linalg/dense.jl
@@ -768,12 +768,7 @@ function factorize(A::StridedMatrix{T}) where T
             return UpperTriangular(A)
         end
         if herm
-            cf = cholfact(A)
-            if cf.info == 0
-                return cf
-            else
-                return factorize(Hermitian(A))
-            end
+            return factorize(Hermitian(A))
         end
         if sym
             return factorize(Symmetric(A))
diff --git a/base/linalg/symmetric.jl b/base/linalg/symmetric.jl
index c757ebf..7aab223 100644
--- a/base/linalg/symmetric.jl
+++ b/base/linalg/symmetric.jl
@@ -309,7 +309,14 @@ for T in (:Symmetric, :Hermitian), op in (:+, :-, :*, :/)
 end
 
 bkfact(A::HermOrSym) = bkfact(A.data, Symbol(A.uplo), issymmetric(A))
-factorize(A::HermOrSym) = bkfact(A)
+function factorize(A::HermOrSym)
+    C = cholfact(A)
+    if isposdef(C)
+        return C
+    else
+        return bkfact(A)
+    end
+end
 
 det(A::RealHermSymComplexHerm) = real(det(bkfact(A)))
 det(A::Symmetric{<:Real}) = det(bkfact(A))
```
which would not change anything if the input is a `Matrix`, but instead return `Cholesky` if the input is `Hermitian` and positive definite.